cli/internal/run/run.go

```diff
@@ -77,128 +78,149 @@ func (rs *runSpec) ArgsForTask(task string) []string {
 	return passThroughArgs
 }
 
+var _cmdLong = `
+Run tasks across projects in your monorepo.
+
+By default, turbo executes tasks in topological order (i.e.
+dependencies first) and then caches the results. Re-running commands for
+tasks already in the cache will skip re-execution and immediately move
+artifacts from the cache into the correct output folders (as if the task
+occurred again).
+
+Arguments passed after '--' will be passed through to the named tasks.
+`
+
+func getCmd(config *config.Config, ui cli.Ui, processes *process.Manager) *cobra.Command {
+	var opts *Opts
+	cmd := &cobra.Command{
+		Use:                   "turbo run <task> [...<task>] [<flags>] -- <args passed to tasks>",
+		Short:                 "Run tasks across projects in your monorepo",
+		Long:                  _cmdLong,
+		SilenceUsage:          true,
+		SilenceErrors:         true,
+		DisableFlagsInUseLine: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			tasks, passThroughArgs := parseTasksAndPassthroughArgs(args, cmd.Flags())
+			if len(tasks) == 0 {
+				return errors.New("at least one task must be specified")
+			}
+			opts.runOpts.passThroughArgs = passThroughArgs
+			run := configureRun(config, ui, opts, processes)
+			return run.run(tasks)
+		},
+	}
+	opts = optsFromFlags(cmd.Flags(), config)
+	return cmd
+}
+
+func parseTasksAndPassthroughArgs(remainingArgs []string, flags *pflag.FlagSet) ([]string, []string) {
+	if argSplit := flags.ArgsLenAtDash(); argSplit != -1 {
+		return remainingArgs[:argSplit], remainingArgs[argSplit:]
+	}
+	return remainingArgs, nil
+}
+
+func optsFromFlags(flags *pflag.FlagSet, config *config.Config) *Opts {
+	opts := getDefaultOptions(config)
+	aliases := make(map[string]string)
+	scope.AddFlags(&opts.scopeOpts, flags)
+	addRunOpts(&opts.runOpts, flags, aliases)
+	noopPersistentOptsDuringMigration(flags)
+	// TODO: this will probably have to change when we are all-cobra and might not
+	// have Cwd yet.
+	cache.AddFlags(&opts.cacheOpts, flags, config.Cwd)
+	runcache.AddFlags(&opts.runcacheOpts, flags)
+	flags.SetNormalizeFunc(func(f *pflag.FlagSet, name string) pflag.NormalizedName {
+		if alias, ok := aliases[name]; ok {
+			return pflag.NormalizedName(alias)
+		}
+		return pflag.NormalizedName(name)
+	})
+	return opts
+}
+
+func configureRun(config *config.Config, output cli.Ui, opts *Opts, processes *process.Manager) *run {
+	if os.Getenv("TURBO_FORCE") == "true" {
```

Comment:
Not for now, but at some point we should make our control via ENV var a bit more reasonable.

Reply:
Can I interest you in [viper](https://github.com/spf13/viper#what-is-viper)?

Reply:
I'm less and less impressed with the spf13 stuff—it all feels like abandonware. afero has burned me. The health of the projects are basically, "who else has contributor privs?"

- [Cobra contributions](https://github.com/spf13/cobra/graphs/contributors?from=2020-06-06&to=2022-06-06&type=c)
- [Viper contributions](https://github.com/spf13/viper/graphs/contributors?from=2020-06-06&to=2022-06-06&type=c)

It appears that `viper` is actively maintained by somebody who actually cares about it, which is great! I'm interested!